packages/dd-trace/src/instrumenter.js

```diff
@@ -16,6 +16,7 @@ class Instrumenter {
     this._names = new Set()
     this._plugins = new Map()
     this._instrumented = new Map()
+    this._disabledPlugins = new Set()
```

Comment:
This could come directly from the config.

Reply:
The reason I'm persisting `_disabledPlugins` in the constructor is because `.use` doesn't have access to the global configuration, and it calls `_set()` directly to set/load a plugin. So, in the case where `.enable` (which does have access to the global configuration) is invoked and then `.use` is later invoked, it would cause a plugin previously disabled to be enabled, unless `_disabledPlugins` are persisted and then validated in `_set()` . This is to ensure your earlier feedback here is adhered to:

> This check should be done directly in `_set` so that calls to `tracer.use()` don't end up enabling the integration by accident.
(https://github.com/DataDog/dd-trace-js/pull/733#discussion_r342180587)

Reply:
I would say in that case to skip Config completely and use the environment variable in the instrumenter. This will avoid having to enable and then disable plugins in that case. It will also simplify the code. We'll revisit the in-code config when we work on the `plugin` option.

Reply:
gotcha, will do, thanks for the feedback should be able to push up something to handle this soon

Reply:
updated accordingly, tried to keep it as simple as possible and document appropriately